packages/components/src/components/chat/Chat.vue

```diff
@@ -115,32 +149,30 @@ export default {
     },
   },
   methods: {
-    addToken(token: string, messageId: string) {
-      let systemMessage = this.messages.find((m) => m.id === messageId);
-      if (!systemMessage) {
-        systemMessage = this.addMessage(false);
-        systemMessage.id = messageId;
+    // Creates-or-appends a message.
+    addToken(token: string, threadId: string, messageId: string) {
+      if (threadId !== this.threadId) return;
+
+      let assistantMessage = this.messages.find((m) => m.messageId === messageId);
+      if (!assistantMessage) {
+        assistantMessage = new AssistantMessage(messageId);
+        this.messages.push(assistantMessage);
       }
-      systemMessage.message += token;
+
+      assistantMessage.append(token);
       this.scrollToBottom();
     },
     setAuthorized(v: boolean) {
       this.authorized = v;
     },
-    addMessage(isUser: boolean, content?: string, isError?: boolean) {
-      const message = {
-        id: this.pendingMessageId++,
-        isUser,
-        isError,
-        message: content || '',
-        sentiment: 0,
-      };
-      this.messages.push(message);
-
+    // Creates a new message, from the user or from the AI.
+    addUserMessage(content: string) {
+      this.messages.push(new UserMessage(content));
+      this.scrollToBottom();
+    },
```

Comment:
Is this comment accurate? It looks like this is only for user messages.

Reply:
Yeah, it looks like it's only for User messages.